### PR TITLE
Fix addJs/addCss on migrated pages with Symfony layout

### LINKED ISSUE #36376: Methods addCSS / addJS not working on migrated controller when used in displayBackOfficeHeader or actionAdminControllerSetMedia hooks
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Hello,

We noticed that when we are using `displayBackOfficeHeader` or `actionAdminControllerSetMedia` hooks, the classic `addCSS` & `addJS` functions aren't working as expected.

To be exact, they are properly called from the `LegacyControllerContext` but aren't transferred to the `LegacyControllerContextProxy*` instance where `js_files` & `css_files` attributes are left empty.

### Expected behavior

`js_files` & `css_files` attributes should be filled in the `LegacyControllerContextProxy*` instance

### Steps to reproduce

1. Install any module that uses `displayBackOfficeHeader` or `actionAdminControllerSetMedia` hooks like ps_faviconnotificationbo
2. Notice that JS assets aren't loaded at all => favico.js and ps_faviconnotificationbo.js
3. Go to any non-migrated controller, like "AdminSearchConf", notice that the assets are loaded.

### PrestaShop version(s) where the bug happened

9.0.0-alpha1

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

PrestaModule

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Split LegacyControllerContextBuilder so the part that build legacy context is called later and uses the same lazy service, this way we make sure migrated pages have the correct assets from legacy modules
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Go to the back office on a migrated page, check that the assets of the `ps_faviconnotificationbo` module are loaded in the page (ex: `/modules/ps_faviconnotificationbo/views/js/ps_faviconnotificationbo.js`) Without the PR the files are not included/loaded in the page With this PR they are present
| UI Tests          | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/9682990461
| Fixed issue or discussion?     | Fixes #36376
| Related PRs       | ~
| Sponsor company   | ~

### CODE REVIEW

**Review by boherm — APPROVED** [review]

**Comment by hadjedjvincent** [comment]
Tested OK on my side

**Review by mflasquin — APPROVED** [review]

**Comment by paulnoelcholot** [comment]
Hello @jolelievre,

I tested your PR and it's good for me! :tada: 

Thanks!

**Review by paulnoelcholot — APPROVED** [review]

**Comment by ps-jarvis** [comment]
QA approved, well done! Message to the maintainers: do not forget to milestone it before the merge.

**Comment by kreare** [comment]
Any chance to get this merged on v8.1 branch? Seems to be merged only on develop branch (thus, for v9?)

**Comment by kpodemski** [comment]
@kreare this problem has been reported on PrestaShop 9, not on 8.1, do you have the same issue on 8.1? also, on v9 quite a lot has happened in the back office and how its layout is rendered and some of the code you see in this PR is not even available in v8.1

**Comment by kreare** [comment]
Yes, on v8.1 it's the same (probably due to the new product page layout)
I've created a custom module that inject a tinymce config, the js is loaded (i can see it from the console) but tinymce doesn't change at all.

Just to be 100% sure: do you have a working example that adds a style (like Header 7) in the paragraph dropdown ?
I've struggled for days trying to get this working and then I came here in this bug report

**Comment by kreare** [comment]
any update on this?

**Comment by jolelievre** [comment]
This topic is fixed and closed

The problem you mention doesn't seem related at all, so if you have an issue and are looking for a solution you should open a dedicated issue and explain it because it's not clear right now what your actual problem is

**Comment by kreare** [comment]
not related? the the same exact issue happening on v8.1 too

**Pull request merged**
